src/androidInstrumentedTest/kotlin/software/momento/kotlin/sdk/CacheClientControlTest.kt

```diff
@@ -44,4 +45,21 @@ class CacheClientControlTest: BaseAndroidTestClass() {
             assert((deleteResponse as CacheDeleteResponse.Error).cause is NotFoundException)
         }
     }
+
+    @Test
+    fun listCacheHappyPath() = runTest  {
+        val cacheName = "kotlin-android-create-delete-${UUID.randomUUID()}"
+        var createResponse = cacheClient.createCache(cacheName)
+        assert(createResponse is CacheCreateResponse.Success)
+
+        var listCachesResponse = cacheClient.listCaches()
+        assert(listCachesResponse is CacheListResponse.Success)
+
+        val caches = (listCachesResponse as CacheListResponse.Success).caches
+        val cacheNames = caches.map { it.name }
+        assert(cacheNames.contains(cacheName))
+
+        var deleteResponse = cacheClient.deleteCache(cacheName)
```

Comment:
Can you do this in a finally block? The cache won't be deleted if there is a transient error with list caches.